src/admin/content-block/helpers/parsers.ts

```diff
@@ -0,0 +1,62 @@
+import { get } from 'lodash-es';
+
+import { CONTENT_BLOCK_CONFIG_MAP } from '../content-block.const';
+import { ContentBlockConfig, ContentBlockSchema, ContentBlockType } from '../content-block.types';
+
+// Parse content-block config to valid request body
+export const parseContentBlockConfig = (
+	contentBlockConfig: ContentBlockConfig,
+	contentId?: number
+) => {
+	const componentState = contentBlockConfig.components.state;
+	const { blockType, position, ...blockState } = contentBlockConfig.block.state;
+
+	return {
+		position,
+		variables: { componentState, blockState },
+		...(contentId ? { content_id: contentId } : null),
+		content_block_type: blockType,
+	};
+};
+
+export const parseContentBlockConfigs = (
+	contentId: number,
+	contentBlockConfigs: ContentBlockConfig[]
+): Partial<ContentBlockSchema>[] => {
+	const contentBlocks = contentBlockConfigs.map(contentBlockConfig =>
+		parseContentBlockConfig(contentBlockConfig, contentId)
+	);
+
+	return contentBlocks;
+};
+
+// Parse content-blocks to configs
+export const parseContentBlocks = (contentBlocks: ContentBlockSchema[]): ContentBlockConfig[] => {
+	const sortedContentBlocks = contentBlocks.sort(
+		(a, b) => (a.position as number) - (b.position as number)
```

Comment:
het lijkt erop dat position altijd een number is. Mss ff verifieren dat op alle locaties waar je een ContentBlockSchema aanmaakt. Indien dit zo is pas je wss de interface van de block best aan:

van:
```typescript
export type ContentBlockSchema = {
	id: number;
	content_id: number;
	variables: { [key: string]: any } | any[] | null;
	position: number | null;
	created_at: string;
	updated_at: string;
	content_block_type: string;
};
```

naar:
```typescript
export type ContentBlockSchema = {
	id: number;
	content_id: number;
	variables: { [key: string]: any } | any[] | null;
	position: number;
	created_at: string;
	updated_at: string;
	content_block_type: string;
};
```

dan kan je alle .position as number castings verwijderen

Reply:
Yes dat zou idd het probleem van het casten verhelpen maar die `position: number | null` is gwn wat er in graphql mogelijk is van values.
Is het dan niet beter om zoiets te doen: `position === null ? blocks.length : position`

Reply:
het is wss best om aan bart te vragen dat hij die tabel aanpast zodat position niet null mag zijn.

Reply:
Added TODO's and will be fixed in typings